### PR TITLE
gui: Reset warnings about nested folder roots (fixes #3433)

### LINKED ISSUE #3433: gui: Warning "path is a subdirectory of other folder" persists between dialogs
Seen in ubuntu firefox
Starting to create a new share, then giving a path through a yet shared one the warning (string 139) triggers OK. Then cancelling the creation and immediately starting a new creation, the warning still shows even though the path is empty. The warning definitively disappears OK when typing / ~ or double-clic the path field... until we type again a path through an existing share path, which is OK.

Seems to be some variable not reset by Close button. Same issue when validating with Save.

### PR DESCRIPTION
**Pull request by imsodin**
Refer to #3433 for a description of the problem.

To reset the warning on modal open the variables set in the `pathIsSubDir`-directive need to be reset. To make this easier I collected them in one object called `folderPathErrors`.

I also found one small mistake in the help text about tilde expansion (path wasn't included in the ng-if condition). And indentation polish.

### CODE REVIEW

File: gui/default/syncthing/core/syncthingController.js [hunk 1]
```diff
@@ -53,27 +53,30 @@ angular.module('syncthing.core')
...
-                simpleKeep: 5,
-                staggeredMaxAge: 365,
-                staggeredCleanInterval: 3600,
-                staggeredVersionsPath: "",
-                externalCommand: "",
-                autoNormalize: true
+            selectedDevices: {},
+            type: "readwrite",
+            rescanIntervalS: 60,
+            fsNotifications: $scope.fsNotificationsAvailable,
```
Comment (AudriusButkevicius) [inline gui/default/syncthing/core/syncthingController.js]: Hmm

Comment (imsodin) [inline gui/default/syncthing/core/syncthingController.js]: My git workspace seems to be a bit messy - thanks for noticing.

**Comment by AudriusButkevicius** [comment]
@st-review merge

**Comment by st-review** [comment]
:ok_hand: Merged as dd6bb6d5fdd01882078d2ef1e57ca74a91daf217. Thanks, @imsodin!

**Pull request closed without merging**